Traceback (most recent call last):
  File "/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/matmek4270/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
r, E, dt = convergence_rates(5)
plt.loglog(dt, E)
plt.loglog(dt, 10*dt)
plt.loglog(dt, np.array(dt)**2)
plt.title('Convergence of finite difference method')
plt.legend(['Error', '$\\mathcal{O}(\\Delta t)$', '$\\mathcal{O}(\\Delta t^2)$'])
from plotslopes import slope_marker
slope_marker((dt[1], E[1]), (2,1))
slope_marker((dt[1], 10*dt[1]), (1,1))
slope_marker((dt[1], dt[1]**2), (2,1))
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[5], line 3[0m
[1;32m      1[0m r, E, dt [38;5;241m=[39m convergence_rates([38;5;241m5[39m)
[1;32m      2[0m plt[38;5;241m.[39mloglog(dt, E)
[0;32m----> 3[0m [43mplt[49m[38;5;241;43m.[39;49m[43mloglog[49m[43m([49m[43mdt[49m[43m,[49m[43m [49m[38;5;241;43m10[39;49m[38;5;241;43m*[39;49m[43mdt[49m[43m)[49m
[1;32m      4[0m plt[38;5;241m.[39mloglog(dt, np[38;5;241m.[39marray(dt)[38;5;241m*[39m[38;5;241m*[39m[38;5;241m2[39m)
[1;32m      5[0m plt[38;5;241m.[39mtitle([38;5;124m'[39m[38;5;124mConvergence of finite difference method[39m[38;5;124m'[39m)

File [0;32m/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/matplotlib/pyplot.py:3398[0m, in [0;36mloglog[0;34m(*args, **kwargs)[0m
[1;32m   3396[0m [38;5;129m@_copy_docstring_and_deprecators[39m(Axes[38;5;241m.[39mloglog)
[1;32m   3397[0m [38;5;28;01mdef[39;00m [38;5;21mloglog[39m([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs) [38;5;241m-[39m[38;5;241m>[39m [38;5;28mlist[39m[Line2D]:
[0;32m-> 3398[0m     [38;5;28;01mreturn[39;00m [43mgca[49m[43m([49m[43m)[49m[38;5;241;43m.[39;49m[43mloglog[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/matplotlib/axes/_axes.py:1865[0m, in [0;36mAxes.loglog[0;34m(self, *args, **kwargs)[0m
[1;32m   1861[0m dy [38;5;241m=[39m {k: v [38;5;28;01mfor[39;00m k, v [38;5;129;01min[39;00m kwargs[38;5;241m.[39mitems()
[1;32m   1862[0m       [38;5;28;01mif[39;00m k [38;5;129;01min[39;00m [[38;5;124m'[39m[38;5;124mbase[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124msubs[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mnonpositive[39m[38;5;124m'[39m,
[1;32m   1863[0m                [38;5;124m'[39m[38;5;124mbasey[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124msubsy[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mnonposy[39m[38;5;124m'[39m]}
[1;32m   1864[0m [38;5;28mself[39m[38;5;241m.[39mset_yscale([38;5;124m'[39m[38;5;124mlog[39m[38;5;124m'[39m, [38;5;241m*[39m[38;5;241m*[39mdy)
[0;32m-> 1865[0m [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mplot[49m[43m([49m
[1;32m   1866[0m [43m    [49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43m{[49m[43mk[49m[43m:[49m[43m [49m[43mv[49m[43m [49m[38;5;28;43;01mfor[39;49;00m[43m [49m[43mk[49m[43m,[49m[43m [49m[43mv[49m[43m [49m[38;5;129;43;01min[39;49;00m[43m [49m[43mkwargs[49m[38;5;241;43m.[39;49m[43mitems[49m[43m([49m[43m)[49m[43m [49m[38;5;28;43;01mif[39;49;00m[43m [49m[43mk[49m[43m [49m[38;5;129;43;01mnot[39;49;00m[43m [49m[38;5;129;43;01min[39;49;00m[43m [49m[43m{[49m[38;5;241;43m*[39;49m[43mdx[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[43mdy[49m[43m}[49m[43m}[49m[43m)[49m

File [0;32m/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/matplotlib/axes/_axes.py:1724[0m, in [0;36mAxes.plot[0;34m(self, scalex, scaley, data, *args, **kwargs)[0m
[1;32m   1481[0m [38;5;250m[39m[38;5;124;03m"""[39;00m
[1;32m   1482[0m [38;5;124;03mPlot y versus x as lines and/or markers.[39;00m
[1;32m   1483[0m 
[0;32m   (...)[0m
[1;32m   1721[0m [38;5;124;03m(``'green'``) or hex strings (``'#008000'``).[39;00m
[1;32m   1722[0m [38;5;124;03m"""[39;00m
[1;32m   1723[0m kwargs [38;5;241m=[39m cbook[38;5;241m.[39mnormalize_kwargs(kwargs, mlines[38;5;241m.[39mLine2D)
[0;32m-> 1724[0m lines [38;5;241m=[39m [[38;5;241m*[39m[38;5;28mself[39m[38;5;241m.[39m_get_lines([38;5;28mself[39m, [38;5;241m*[39margs, data[38;5;241m=[39mdata, [38;5;241m*[39m[38;5;241m*[39mkwargs)]
[1;32m   1725[0m [38;5;28;01mfor[39;00m line [38;5;129;01min[39;00m lines:
[1;32m   1726[0m     [38;5;28mself[39m[38;5;241m.[39madd_line(line)

File [0;32m/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/matplotlib/axes/_base.py:303[0m, in [0;36m_process_plot_var_args.__call__[0;34m(self, axes, data, *args, **kwargs)[0m
[1;32m    301[0m     this [38;5;241m+[39m[38;5;241m=[39m args[[38;5;241m0[39m],
[1;32m    302[0m     args [38;5;241m=[39m args[[38;5;241m1[39m:]
[0;32m--> 303[0m [38;5;28;01myield from[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_plot_args[49m[43m([49m
[1;32m    304[0m [43m    [49m[43maxes[49m[43m,[49m[43m [49m[43mthis[49m[43m,[49m[43m [49m[43mkwargs[49m[43m,[49m[43m [49m[43mambiguous_fmt_datakey[49m[38;5;241;43m=[39;49m[43mambiguous_fmt_datakey[49m[43m)[49m

File [0;32m/opt/anaconda3/envs/matmek4270/lib/python3.12/site-packages/matplotlib/axes/_base.py:499[0m, in [0;36m_process_plot_var_args._plot_args[0;34m(self, axes, tup, kwargs, return_kwargs, ambiguous_fmt_datakey)[0m
[1;32m    496[0m     axes[38;5;241m.[39myaxis[38;5;241m.[39mupdate_units(y)
[1;32m    498[0m [38;5;28;01mif[39;00m x[38;5;241m.[39mshape[[38;5;241m0[39m] [38;5;241m!=[39m y[38;5;241m.[39mshape[[38;5;241m0[39m]:
[0;32m--> 499[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mx and y must have same first dimension, but [39m[38;5;124m"[39m
[1;32m    500[0m                      [38;5;124mf[39m[38;5;124m"[39m[38;5;124mhave shapes [39m[38;5;132;01m{[39;00mx[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m and [39m[38;5;132;01m{[39;00my[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m"[39m)
[1;32m    501[0m [38;5;28;01mif[39;00m x[38;5;241m.[39mndim [38;5;241m>[39m [38;5;241m2[39m [38;5;129;01mor[39;00m y[38;5;241m.[39mndim [38;5;241m>[39m [38;5;241m2[39m:
[1;32m    502[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mx and y can be no greater than 2D, but have [39m[38;5;124m"[39m
[1;32m    503[0m                      [38;5;124mf[39m[38;5;124m"[39m[38;5;124mshapes [39m[38;5;132;01m{[39;00mx[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m and [39m[38;5;132;01m{[39;00my[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m"[39m)

[0;31mValueError[0m: x and y must have same first dimension, but have shapes (5,) and (50,)

